osquery/logger/plugins/aws_firehose.cpp

```diff
@@ -55,20 +55,43 @@ Status FirehoseLoggerPlugin::logString(const std::string& s) {
   return forwarder_->logString(s);
 }
 
+Status FirehoseLoggerPlugin::logStatus(const std::vector<StatusLogLine>& log) {
+  return forwarder_->logStatus(log);
+}
+
+void FirehoseLoggerPlugin::init(const std::string& name,
+                                const std::vector<StatusLogLine>& log) {
+  google::ShutdownGoogleLogging();
+  google::InitGoogleLogging(name.c_str());
+  logStatus(log);
+}
+
 Status FirehoseLogForwarder::send(std::vector<std::string>& log_data,
                                   const std::string& log_type) {
   std::vector<Aws::Firehose::Model::Record> records;
   for (const std::string& log : log_data) {
-    if (log.size() + 1 > kFirehoseMaxLogBytes) {
+    std::string typed_log;
+
+    Status status = appendLogTypeToJson(log_type, log, typed_log);
```

Comment:
In this instance we could. However, we would need to slightly modify `appendLogTypeToJson`. The Kinesis logger plugin, on failure to send, will use the item from `log_data` and put it in the processing loop again to be sent. If we mutate `log_data`'s contents, we would need some logic to prevent appending the `log_type` key on the second pass...